app/src/main/java/io/gnosis/safe/ui/transactions/details/TransactionDetailsViewModel.kt

```diff
@@ -34,6 +37,20 @@ class TransactionDetailsViewModel
                 true == ownerCredentialsRepository.retrieveCredentials()?.let { credentials ->
             executionInfo.signers.contains(credentials.address) && !executionInfo.confirmations.map { it.signer }.contains(credentials.address)
         }
+
+    suspend fun validateSafeTxHash(transaction: TransactionDetails): Boolean {
+        kotlin.runCatching {
+            val safe = safeRepository.getActiveSafe()
+            val safeTxHash = (transaction.detailedExecutionInfo as DetailedExecutionInfo.MultisigExecutionDetails).safeTxHash
```

Comment:
if The only thing you need from the `MultisigExecutionDetails` is the `safeTxHash`, I would say you modify this method to require the minimum information necessary for that. Meaning, the `safeTxHash: String` directly. (or whichever type that is).

That way you can get rid of the hard cast.

Reply:
we'll have to do hard cast at the upper level. I think having it encapsulated in validateSafeTxHash is more elegant than doing it at the viewModel level

Reply:
besides executionInfo is already part of the transaction object

Reply:
The only subtype of `DetailedExecutionInfo` that has a `safeTxHash` is the  `MultisigExecutionDetails`, so I would say it would be far more elegant to leverage the strong typing of kotlin to your advantage. 

I don't think that the cast is necessary at all, as this callback is probably triggered from a context the type is already known ? I am imagining that it comes from the big `when` block for `executionInfo`

Reply:
you proposing passing both, transaction and executionInfo as parameters, right?

Reply:
Yeah, just passing both. Or even you could create a specific structure for this, if you prefer. because we are not taking advantage of typing here. If we specificially use `MultisigExecutionDetails` it is better documented that this method only works for these types. If you feel this is too knit-picky we can drop it, but I think that we should use types when we have them for the sake of clarity.

Reply:
Please recheck the latest state